src/WebJobs.Extensions.ApiHub/Listener/ApiHubListener.cs

```diff
@@ -57,13 +57,11 @@ public Task StartAsync(CancellationToken cancellationToken)
 
             if (_poll == null)
             {
-                var errorText = string.Format("Path '{0}' is invalid. IFolderItem.RootPath must be set to a valid directory location.", _folderSource.Path);
-                _trace.Error(errorText);
-
-                throw new InvalidOperationException(errorText);
+                var errorText = string.Format("Path '{0}' is invalid. Path must be set to a valid directory location.", _folderSource.Path);
+                var ex = new InvalidOperationException(errorText);
+                _trace.Error(errorText, ex);
```

Comment:
There is no portal or anything in the context of this repo :) This extension should be designed to stand alone in a regular SDK WebJob, outside of Functions. The right behavior in that context is to bring down the host if there are any errors in starting up the job methods.

I think the right thing for us to do is make a change to the core SDK to allow listener exceptions to be "handled". I did something similar for indexing exceptions (in this [commit](https://github.com/Azure/azure-webjobs-sdk/commit/4842e35d5b6b780904ec484b46b8aea063fd566b)), to allow the Functions runtime to mark indexing exceptions as "handled" - they normally bring down the host. This listener case is similar I think.